libclc/utils/libclc-remangler/LibclcRemangler.cpp

```diff
@@ -147,340 +123,606 @@ class BumpPointerAllocator {
   }
 
   ~BumpPointerAllocator() { reset(); }
+
+private:
+  void grow() {
+    char *NewMeta = static_cast<char *>(std::malloc(AllocSize));
+    if (NewMeta == nullptr)
+      std::terminate();
+    BlockList = new (NewMeta) BlockMeta{BlockList, 0};
+  }
+
+  void *allocateMassive(size_t NBytes) {
+    NBytes += sizeof(BlockMeta);
+    BlockMeta *NewMeta = reinterpret_cast<BlockMeta *>(std::malloc(NBytes));
+    if (NewMeta == nullptr)
+      std::terminate();
+    BlockList->Next = new (NewMeta) BlockMeta{BlockList->Next, 0};
+    return static_cast<void *>(NewMeta + 1);
+  }
+
+private:
+  struct BlockMeta {
+    BlockMeta *Next;
+    size_t Current;
+  };
+
+  static constexpr size_t AllocSize = 4096;
+  static constexpr size_t UsableAllocSize = AllocSize - sizeof(BlockMeta);
+
+  alignas(long double) char InitialBuffer[AllocSize];

```

Comment:
I know this is existing code, but... why aligning to `long double`?

Reply:
I'm not sure, but am guessing the largest scalar type, with `long` being a nod towards windows, to make sure that it does get 64-bit (https://learn.microsoft.com/en-us/cpp/cpp/fundamental-types-cpp?view=msvc-170#floating-point-types):

>long double and double are treated as distinct types by the compiler. The Microsoft C++ compiler uses the 4- and 8-byte IEEE-754 floating-point representations

Reply:
Done